### PR TITLE
TiFlash: Fix multi-dir mkdir failure

### PR DESCRIPTION
**Pull request by birdstorm**
When multi directory exists in data-dir config, it should behave normally with mkdir command.

### CODE REVIEW

File: cmd/cluster/command/deploy.go [hunk 1]
```diff
@@ -228,7 +232,7 @@ func deploy(clusterName, clusterVersion, topoFile string, opt deployOptions) err
 				globalOptions.User,
 				meta.DirPaths{
 					Deploy: deployDir,
-					Data:   dataDir,
+					Data:   strings.Join(dataDirs, " "),
```
Comment (lonng) [inline cmd/cluster/command/deploy.go:235]: I don't think it's correct to concatenate all data directories with space.

**Comment by AstroProfundis** [comment]
Maybe just change `DataDir string` to `DataDir []string` for all spec structs?

File: cmd/cluster/command/scale_out.go [hunk 2]
```diff
@@ -200,18 +203,22 @@ func buildScaleOutTask(
...
 		tb := task.NewBuilder().
 			UserSSH(inst.GetHost(), inst.GetSSHPort(), metadata.User, sshTimeout).
 			Mkdir(metadata.User, inst.GetHost(),
-				deployDir, dataDir, logDir,
+				deployDir, logDir,
 				filepath.Join(deployDir, "bin"),
 				filepath.Join(deployDir, "conf"),
-				filepath.Join(deployDir, "scripts"))
+				filepath.Join(deployDir, "scripts")).
+			Mkdir(metadata.User, inst.GetHost(), dataDirs...)
```
Comment (AstroProfundis) [inline cmd/cluster/command/scale_out.go:221]: I believe it's ok to append `dataDirs...` to the `Mkdir` task, this would avoid one extra SSH connection

Comment (birdstorm) [reply cmd/cluster/command/scale_out.go:221]: that is not possible, golang error goes: Invalid use of ..., corresponding parameter is non-variadic

**Comment by birdstorm** [comment]
@AstroProfundis comments addressed. ptal

**Comment by lonng** [comment]
@birdstorm I think you should check the `len(c.path.Data) > 0` in the function https://github.com/pingcap-incubator/tiup-cluster/blob/c9eb8cb5a9b51e849e3b18f41ec18308a9679048/pkg/task/init_config.go#L34.

**Comment by AstroProfundis** [comment]
Does the new schema compatible with clusters deployed by older releases? (and also imported clusters)

**Comment by birdstorm** [comment]
@lonng The result of strings.Split is a slice with one element - the empty string. Which means paths.Data has at least one element.

**Comment by birdstorm** [comment]
@AstroProfundis This PR fixes the incorrect parsing of multi-directory settings, which created an incorrect AND un-used directory.

File: cmd/cluster/command/check.go [hunk 3]
```diff
@@ -155,7 +155,7 @@ func checkSystemInfo(s *cliutil.SSHConnectionProps, topo *meta.TopologySpecifica
 			// if the data dir set in topology is relative, and the home dir of deploy user
 			// and the user run the check command is on different partitions, the disk detection
 			// may be using incorrect partition for validations.
-			dataDir := clusterutil.Abs(opt.user, inst.DataDir())
+			dataDir := clusterutil.Abs(opt.user, strings.Split(inst.DataDir(), ",")[0])
```
Comment (lonng) [inline cmd/cluster/command/check.go:158]: I don't think it enough just checks the first directory if the component if TiFlash.

File: cmd/cluster/command/upgrade.go [hunk 4]
```diff
@@ -110,7 +111,10 @@ func upgrade(clusterName, clusterVersion string, opt operator.Options) error {
 
 			deployDir := clusterutil.Abs(metadata.User, inst.DeployDir())
 			// data dir would be empty for components which don't need it
-			dataDir := clusterutil.Abs(metadata.User, inst.DataDir())
+			var dataDirs []string
+			for _, dataDir := range strings.Split(inst.DataDir(), ",") {
+				dataDirs = append(dataDirs, clusterutil.Abs(metadata.User, dataDir))
+			}
```
Comment (lonng) [inline cmd/cluster/command/upgrade.go:117]: Can we extract a function for them? It's duplicated in some places.

Comment (birdstorm) [reply cmd/cluster/command/upgrade.go:117]: sure.

**Pull request merged**